.github/workflows/test-builds.yml

```diff
@@ -180,3 +180,47 @@ jobs:
     - name: Run ${{ matrix.test-platform }} Tests
       run: |
         make tests-single-ci TEST_TARGET=${{ matrix.test-platform }}
+
+  # This runs all unit tests as a single job. While it should be possible to break this up into
+  # multiple jobs, they currently run quickly and finish long before the compilation tests.
+  run_unit_tests:
+    name: Run All Unit Tests
+    # These tests will only be able to run on the bugfix-2.1.x branch, until the next release
+    # pulls them into additional branches.
+    if: github.repository == 'MarlinFirmware/Marlin' && github.ref == 'refs/heads/bugfix-2.1.x'
```

Comment:
While I tested that the workflow worked on my own repo/branch, I didn't test this `github.ref` check. I'm attempting to prevent this from running on the 2.1.x branch, since it won't work yet.

If we don't want to gamble whether this will work once merged, we could split this into a new YML file with different branch requirements. I thought it made sense to add it to this file because then it can re-use all the trigger configuration regarding file paths.

Reply:
That github.ref syntax came from GitHub Copilot.

Reply:
We could also move this to its own action, if that helps to make things cleaner.